### DISCUSSION TITLE
R4 display, adapter with external antenna

### DISCUSSION THREAD

**Discussion by K1LBG** [WLAN-Pi/feedback#43] [General Inquiries] [answered]
Hi, looking to put together a wifi analyzer.  However, I want to use a directional antenna, thus an adapter with an antenna connection.  Also, while I could live with  an output list that shows AP/device SSID and RSSI, it would be great if I could get a graphical output.  Any thoughts on either?  Thanks in advance.

**Comment by jolla** [comment]
It sounds like you are looking for a USB Wi-Fi adapter with external antenna ports? I would recommend checking out Alfa, they make plenty of USB Wi-Fi adapters with RP-SMA connectors.

**Reply by K1LBG** [reply]
Thanks, I guess I'll buy one and try it when I can find a Pi in stock.

**Reply by K1LBG** [reply]
Update:  borrowed a Pi R4 from a friend and burned the latest release to a micro SD.  Also bought an ALFA AWUS036ACS Wi-Fi adapter.  powered on the Pi and the system found an ethernet connection and I can get to it from my pc but it sayw no WI-FI adapter.  Did I buy the wrong Alfa adaper?

**Reply by jolla** [reply]
I'm not familiar with the AWUS036ACS, but it appears to use a Realtek chipset which is likely why it isn't recognized. For the best linux support I'd recommend sticking with adapters that use Mediatek chipsets. For example the AWUS036ACM uses a Mediatek MT7612U chipset or if you have a desire for Wi-Fi 6/6E go with something like the Comfast CF-951 or Netgear A8000 these all use Mediatek chipsets.

**Comment by K1LBG** [comment]
Thanks, I guess I'll buy one and try it when I can find a Pi in stock.

**Comment by jiribrejcha** [comment]
There are some Pis that RPiLocator.com doesn’t track. These are typically kits - bundles of RPi4 and few accessories. 

For example, in the UK, Pimoroni has RPi4 2GB and 4 GB in stock and ready to ship today:
https://shop.pimoroni.com/products/raspberry-pi-4-essentials-kit?variant=40110476230739

**Reply by K1LBG** [reply]
Thanks, I'll look into that.

**Comment by jolla** [comment, marked as answer]
Have a look at these, plenty of options with external antennas.
https://github.com/morrownr/USB-WiFi/blob/main/home/USB_WiFi_Adapters_that_are_supported_with_Linux_in-kernel_drivers.md

**Reply by K1LBG** [reply]
Thanks for the guidance.  I've ordered the original (nondirectional) wifi specified in the build article just to prove everything works.  Once it comes in  and it works, I'll go for one with external connectors (or crack open the original and add connectors).
